CRM/Admin/Form.php

```diff
@@ -70,9 +70,14 @@ public function preProcess() {
     $this->_values = [];
     if (isset($this->_id)) {
       $params = ['id' => $this->_id];
-      // this is needed if the form is outside the CRM name space
-      $baoName = $this->_BAOName;
-      $baoName::retrieve($params, $this->_values);
+      if ($this->getDefaultEntity() === 'PaymentProcessor') {
+        $this->_values = civicrm_api4($this->getDefaultEntity(), 'get', ['where' => [['id', '=', $this->_id]]])->first();
+      }
+      else {
+        // this is needed if the form is outside the CRM name space
+        $baoName = $this->_BAOName;
+        $baoName::retrieve($params, $this->_values);
+      }
```

Comment:
What's going on here @seamuslee001 ?

Reply:
@colemanw so because of the deprecation notice here https://github.com/civicrm/civicrm-core/blob/master/CRM/Financial/BAO/PaymentProcessor.php#L82 I had to write the above code so that the unit test would actually get the point of testing the form function as otherwise it stops when it hits the deprecation notice

Reply:
@seamuslee001 just remove that deprecation notice.
I think we need a different way to roll out this change, I don't want to have to hack in a bunch of form names into this class, and we can't do it blindly because of the subtle differences between `BAO::retrieve` and `APIv4::get` (notably automatic deserialization and data typing done by the api, whereas the `retrieve` function gets everything as a string).

Reply:
ok have done so now @colemanw